src/aks-preview/azext_aks_preview/_loadbalancer.py

```diff
@@ -26,48 +26,58 @@ def set_load_balancer_sku(sku, kubernetes_version):
-    if not is_load_balancer_profile_provided(managed_outbound_ip_count, outbound_ips, outbound_ip_prefixes,
-                                             outbound_ports, idle_timeout):
+    if not is_load_balancer_profile_provided(managed_outbound_ip_count, managed_outbound_ipv6_count, outbound_ips,
+                                             outbound_ip_prefixes, outbound_ports, idle_timeout):
         return profile
-    return configure_load_balancer_profile(managed_outbound_ip_count, outbound_ips, outbound_ip_prefixes,
-                                           outbound_ports, idle_timeout, profile)
+    return configure_load_balancer_profile(managed_outbound_ip_count, managed_outbound_ipv6_count, outbound_ips,
+                                           outbound_ip_prefixes, outbound_ports, idle_timeout, profile)
 
 
-def create_load_balancer_profile(managed_outbound_ip_count, outbound_ips, outbound_ip_prefixes,
-                                 outbound_ports, idle_timeout):
+def create_load_balancer_profile(managed_outbound_ip_count, managed_outbound_ipv6_count, outbound_ips,
+                                 outbound_ip_prefixes, outbound_ports, idle_timeout):
     """parse and build load balancer profile"""
-    if not is_load_balancer_profile_provided(managed_outbound_ip_count, outbound_ips, outbound_ip_prefixes,
-                                             outbound_ports, idle_timeout):
+    if not is_load_balancer_profile_provided(managed_outbound_ip_count, managed_outbound_ipv6_count, outbound_ips,
+                                             outbound_ip_prefixes, outbound_ports, idle_timeout):
         return None
 
     profile = ManagedClusterLoadBalancerProfile()
-    return configure_load_balancer_profile(managed_outbound_ip_count, outbound_ips, outbound_ip_prefixes,
-                                           outbound_ports, idle_timeout, profile)
+    return configure_load_balancer_profile(managed_outbound_ip_count, managed_outbound_ipv6_count, outbound_ips,
+                                           outbound_ip_prefixes, outbound_ports, idle_timeout, profile)
 
 
-def configure_load_balancer_profile(managed_outbound_ip_count, outbound_ips, outbound_ip_prefixes, outbound_ports,
-                                    idle_timeout, profile):
+def configure_load_balancer_profile(managed_outbound_ip_count, managed_outbound_ipv6_count, outbound_ips,
+                                    outbound_ip_prefixes, outbound_ports, idle_timeout, profile):
     """configure a load balancer with customer supplied values"""
     if not profile:
         return profile
 
     outbound_ip_resources = _get_load_balancer_outbound_ips(outbound_ips)
     outbound_ip_prefix_resources = _get_load_balancer_outbound_ip_prefixes(outbound_ip_prefixes)
 
-    if managed_outbound_ip_count or outbound_ip_resources or outbound_ip_prefix_resources:
-        # ips -> i_ps due to track 2 naming issue
-        profile.managed_outbound_i_ps = None
```

Comment:
I am not sure if we need to clear the setting of `managed_outbound_i_ps` when only `outbound_ip_resources` or `outbound_ip_prefix_resources` is specified (in an update scenario).

Reply:
I am not really sure why this is there in the first place. It seems like this has unintended consequences. If I create a cluster then run `az aks update` with outbound_ip_resources, it seems like the managed_outbound_i_ps will be reverted back to the default values which is just `count=1`.